adagucserverEC/Types/ProjectionStore.cpp

```diff
@@ -0,0 +1,73 @@
+/******************************************************************************
+ *
+ * Project:  ADAGUC Server
+ * Purpose:  ADAGUC OGC Server
+ * Author:   Maarten Plieger, plieger "at" knmi.nl
+ * Date:     2024-02-29
+ *
+ ******************************************************************************
+ *
+ * Copyright 2024, Royal Netherlands Meteorological Institute (KNMI)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+#include "ProjectionStore.h"
+#include <bits/pthreadtypes.h>
+#include <pthread.h>
+
+const char *ProjectionStore::className = "ProjectionStore";
+
+ProjectionStore projectionStore;
+extern ProjectionStore projectionStore;
+
+ProjectionStore *ProjectionStore::getProjectionStore() { return &projectionStore; }
+
+ProjectionStore::ProjectionStore() {}
+ProjectionStore::~ProjectionStore() { clear(); }
+
+pthread_mutex_t ProjectionKey_clear;
+void ProjectionStore::clear() { keys.clear(); }
+
+int ProjectionStore::findExtentForKey(ProjectionKey pKey, BBOX *bbox) {
```

Comment:
Why not define a less than operator on pKey, so you can just stuff them in a map?

Reply:
Info:
https://cplusplus.com/forum/general/145841/
https://en.cppreference.com/w/cpp/utility/tuple/make_tuple